source/languages/c/generate.rb

```diff
@@ -1,6 +1,6 @@
-require_relative '../textmate_tools.rb'
+require_relative '../../textmate_tools.rb'
 require_relative './tokens.rb'
-require_relative '../shared/numeric.rb'
+require_relative '../../shared_patterns/numeric.rb'
```

Comment:
with scripts/generate.rb now, would it be better to add /source/ to the load_path and use?:
`require textmate_tools.rb`
`require shared_patterns/numeric.rb`

Reply:
I didn't even know `load_path` existed, thats really cool. I think that'd be a much better solution

Reply:
Well now that I've implemented it, it seems decently harder to understand whats going on
```
$LOAD_PATH << File.join(File.dirname(__FILE__), "../../") # add source to the load path 
require 'textmate_tools.rb'
require 'repo_specific_helpers.rb'
require 'shared_patterns/numeric.rb'
require 'shared_patterns/predefined_macros.rb'
require_relative 'tokens'
```

Do you think this is any better?
```
source_dir = "../../"
require_relative source_dir + 'textmate_tools.rb'
require_relative source_dir + 'repo_specific_helpers.rb'
require_relative source_dir + 'shared_patterns/numeric.rb'
require_relative source_dir + 'shared_patterns/predefined_macros.rb'
require_relative './tokens.rb'
```

Reply:
Yeah, the second version is better.